docs/DEVELOPMENT-GUIDE.md

```diff
@@ -260,12 +260,14 @@ C has a history of issues around memory and buffer handling. To avoid problems i
 
 ### s2n_blob: keeping track of memory ranges
 
-`s2n_blob` is a very simple data structure:
+`s2n_blob` is a data structure representing allocated, "owned" memory or a reference to some other slice of memory.
```

Comment:
I know you later say "or a reference to some other slice of memory", but I worry that someone is still still going to read this as "blobs are always allocated". Maybe just "s2n_blob is a data structure representing a slice of memory"?